realsense/plugin.cpp

```diff
@@ -110,43 +182,72 @@ class realsense : public plugin {
                     using time_point = std::chrono::system_clock::time_point;
                     time_type imu_time_point{std::chrono::duration_cast<time_point::duration>(std::chrono::nanoseconds(imu_time))};
 
-                    // Images
-                    std::optional<cv::Mat *> img0 = std::nullopt;
-                    std::optional<cv::Mat *> img1 = std::nullopt;
-                    std::optional<cv::Mat *> rgb = std::nullopt;
-                    std::optional<cv::Mat *> depth = std::nullopt;
-                    if (last_iteration_cam != cam_type_.iteration)
-                    {
-                        last_iteration_cam = cam_type_.iteration;
-                        img0 = cam_type_.img0;
-                        img1 = cam_type_.img1;
-                        rgb = cam_type_.rgb;
-                        depth = cam_type_.depth;
+                     // Images
+                    if (realsense_cam.compare("D4XX") == 0){
+                        std::optional<cv::Mat *> img0 = std::nullopt;
+                        std::optional<cv::Mat *> img1 = std::nullopt;
+                        std::optional<cv::Mat *> rgb = std::nullopt;
+                        std::optional<cv::Mat *> depth = std::nullopt;
```

Comment:
Having a pointer type inside of a `std::optional` is generally redundant, but the current interface uses `cv::Mat*` for the Switchboard data type. As a heads up, this changes in the upcoming [#32 patch](https://github.com/ILLIXR/ILLIXR/blob/865edb7cfe2e13dce2f0972cd3840c61889afeae/common/data_format.hpp#L33).